drivers/ieee802154/ieee802154_cc13xx_cc26xx_subg.c

```diff
@@ -434,17 +435,27 @@ static void ieee802154_cc13xx_cc26xx_subg_rx_done(
 {
 	struct net_pkt *pkt;
 	uint8_t len;
-	int8_t rssi;
+	int8_t rssi, status;
 	uint8_t *sdu;
 
 	for (int i = 0; i < CC13XX_CC26XX_NUM_RX_BUF; i++) {
 		if (drv_data->rx_entry[i].status == DATA_ENTRY_FINISHED) {
-			len = drv_data->rx_data[i][1];
-			sdu = &drv_data->rx_data[i][3];
-			rssi = sdu[len - 2];
-			len -= 2;
+			len = drv_data->rx_data[i][0];
+			sdu = drv_data->rx_data[i] + 1;
+			status = drv_data->rx_data[i][len--];
+			rssi = drv_data->rx_data[i][len--];
+
+			if (IS_ENABLED(CONFIG_IEEE802154_RAW_MODE)) {
```

Comment:
We should just leverage the `bAppendCrc` bit in `rxconf` and have the radio calculate the crc. I think you mentioned that it's bit-reversed though. Rather than spending a number of CPU cycles re-calculating the CRC, we should just use a bit-reversal algorithm until there is a better solution.

[GCC very badly needs a builtin for doing bit-reversal](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=50481).

I think Zephyr should adopt some convention for it too, so I added #30010.

A simple algorithm would be to use static inline function calls wrapped around a LUT-driven implementation. It would be O(1) rather than O(N).
```cpp
uint8_t bitrev4(uint8_t);
```